### ISSUE TITLE
test: test-crypto-domains should be made non-flaky

### ISSUE THREAD

**Issue by joaocgreis** [joyent/node#25656]
simple/test-crypto-domains is currently marked as flaky. It should be investigated and made non-flaky.

Quick command for testing:
```
time (OK=0; NOK=0; for i in `seq 10000`; do ./out/Release/node test/simple/test-crypto-domains.js && OK=$(($OK+1)) || NOK=$(($NOK+1)); echo "$i   OK: $OK   NOT OK: $NOK"; done)
```

Links:
- File: https://github.com/joyent/node/blob/b8ac6584faf00f6bc9302913ea58e54dcf8f559c/test/simple/test-crypto-domains.js
- File history: https://github.com/joyent/node/commits/b8ac6584faf00f6bc9302913ea58e54dcf8f559c/test/simple/test-crypto-domains.js
- Marked flaky on: [Nov 7, 2014](https://github.com/joyent/node/commit/0674cbaceb81ba1130355aa7eb8e6389824e3cc4)

**Comment by joaocgreis** [comment]
I cannot see this test fail on either Linux x64 or Windows 2012r2 x64. It may have been fixed or affect only certain configurations. Last change to the test was made before it was marked flaky.

**Comment by joaocgreis** [comment]
This test was fixed by 2afa3d8a03f1f0798d83dc57abc252bb78b7e591. Although that commit was already made when the test was marked flaky, it was in a different branch.

Unmarked flaky in 78d256e7f5def6e1ffbf5fb5b232ecf78bc7bde5.
